RUFAS/input_manager.py

```diff
@@ -563,338 +546,364 @@ def _add_default_values_to_array_inputs(
 
         return updated_array, missing_required_property_keys, property_keys_with_default_values
 
-    def _validate_input_type_dynamic(
+    def _validate_input_by_type(
         self,
         variable_properties: Dict[str, Any],
-        var_name: str,
-        input_data_value: Any,
+        variable_path: List[str | int],
+        input_data: Dict[str, Any],
+        eager_termination: bool,
         properties_blob_key: str,
+        elements_counter: "ElementsCounter",
```

Comment:
```suggestion
        elements_counter: ElementsCounter,
```
I think it would be better to move `ElementsCounter` to the top of the file, then type hit it normally.

Reply:
Yes, this would help with typing inside the InputManager class, but within the ElementsCounter class, I would still need to use quotation marks around the type (like the add method).